server/src/main/java/com/linecorp/centraldogma/server/internal/storage/project/DefaultProject.java

```diff
@@ -89,15 +98,19 @@ public class DefaultProject implements Project {
         boolean success = false;
         try {
             createReservedRepos(System.currentTimeMillis());
-            final UserAndTimestamp creation = metadataCreation();
-            if (creation != null) {
+            final ProjectMetadata projectedMetadata = initialMetadata();
+            if (projectedMetadata != null) {
+                final UserAndTimestamp creation = projectedMetadata.creation();
                 creationTimeMillis = creation.timestampMillis();
                 author = Author.ofEmail(creation.user());
+                attachMetadataListener();
             } else {
                 creationTimeMillis = repos.get(REPO_DOGMA).creationTimeMillis();
                 author = repos.get(REPO_DOGMA).author();
             }
             success = true;
+        } catch (Exception e) {
+            throw new CentralDogmaException("failed to initialize internal repositories", e);
```

Comment:
nit:
```suggestion
            throw new CentralDogmaException("failed to initialize internal repositories of " + name, e);
```
Please also add the project name to the exception message below.